AllJoyn/Platform/Bridge/ScriptAdapterLib/ScriptHost.h

```diff
@@ -283,62 +284,110 @@ class ScriptHost
         });
     }
 
-    static void ConvertPropertyValuesToJxValues(const Bridge::AdapterValueVector& valuesIn, std::vector<JXValue>& valuesOut)
+private:
+
+    static void ConvertAdapterValueToJxValue(const std::shared_ptr<Bridge::IAdapterValue>& valueIn, JXValue* valueOut)
+    {
+        Bridge::PropertyValue ipv = (valueIn.get())->Data();
+        switch (ipv.Type())
+        {
+        case Bridge::PropertyType::Boolean:
+            JX_SetBoolean(valueOut, ipv.Get<bool>());
+            break;
+        case Bridge::PropertyType::UInt8:
+            JX_SetInt32(valueOut, ipv.Get<uint8_t>());
+            break;
+        case Bridge::PropertyType::Int16:
+            JX_SetInt32(valueOut, ipv.Get<int16_t>());
+            break;
+        case Bridge::PropertyType::Int32:
+            JX_SetInt32(valueOut, ipv.Get<int32_t>());
+            break;
+        case Bridge::PropertyType::Int64:
+            // Lossy conversion
+            JX_SetInt32(valueOut, ipv.Get<int32_t>());
+            break;
+        case Bridge::PropertyType::UInt16:
+            JX_SetInt32(valueOut, ipv.Get<uint16_t>());
+            break;
+        case Bridge::PropertyType::UInt32:
+            JX_SetInt32(valueOut, ipv.Get<uint32_t>());
+            break;
+        case Bridge::PropertyType::UInt64:
+            // Lossy conversion
+            JX_SetInt32(valueOut, ipv.Get<uint32_t>());
+            break;
+        case Bridge::PropertyType::Double:
+            JX_SetDouble(valueOut, ipv.Get<double>());
+            break;
+        case Bridge::PropertyType::String:
+            JX_SetString(valueOut, ipv.Get<std::string>().c_str(), (const int32_t)ipv.Get<std::string>().length());
+            break;
+        case Bridge::PropertyType::Int32Array:
+        {
+            JX_CreateArrayObject(valueOut);
+            std::vector<int32_t> array = ipv.Get<std::vector<int32_t>>();
+            for (unsigned int i = 0; i < array.size(); i++)
```

Comment:
okk. sure.

---
In reply to: [67948794](https://github.com/tjaffri/msiot-samples/pull/7#discussion_r67948794) [](ancestors = 67948794,67947645)